statsmodels/genmod/generalized_linear_model.py

```diff
@@ -574,10 +584,13 @@ def estimate_scale(self, mu):
         if isinstance(self.scaletype, str):
             if self.scaletype.lower() == 'x2':
                 resid = self.endog - mu
-                return ((np.power(resid, 2) / self.family.variance(mu)).sum()
-                        / self.df_resid)
+                return ((self.freq_weights * (np.power(resid, 2) /
+                         self.family.variance(mu))).sum() /
+                        (self.freq_weights.sum() - self.df_model - 1))
             elif self.scaletype.lower() == 'dev':
-                return self.family.deviance(self.endog, mu)/self.df_resid
+                return (self.family.deviance(self.endog, mu,
+                                             self.freq_weights) /
+                        self.freq_weights.sum() - self.df_model - 1)
```

Comment:
it looks like a missing opening parenthesis in `(self.freq_weights.sum() - self.df_model - 1)`

we can change df_resid to `self.freq_weights.sum() - self.df_model - 1` then these cases with df_resid don't have to be changed, and it's a single point for verifying the definition